src/metabase/pulse/render/body.clj

```diff
@@ -334,6 +387,15 @@
    "#c589b9" "#efce8c" "#b5f95c" "#e35850" "#554dbf" "#bec589" "#8cefc6" "#5cc2f9" "#55e350" "#bf4d4f"
    "#89c3c5" "#be8cef" "#f95cd0" "#50e3ae" "#bf974d" "#899bc5" "#ef8cde" "#f95c67"])
 
+(defn- fill-vector-maybe
+  "You have a sequence of colors or names or whatever,
+  only some of which actually are colors or names and the rest are nil.
+  Fills those nil bits in with the default given seq"
+  [maybe-vector defaults]
+  (let [pairs (map vector maybe-vector defaults)]
```

Comment:
if `defaults` is shorter than the `maybe-vector` this will throw away values. I think you want `(map (fn [v default] (or v default)) maybe-vector (cycle defaults))`